### DISCUSSION TITLE
Login to Keycloak from Websphere to obtain an access_token

### DISCUSSION THREAD

**Discussion by cdecinkoKnight** [keycloak/keycloak#29097] [Q&A]
I am asking this question here as I think I have read everything there is online about the subject.

We have a WebSphere portal that is our main application for our users. They need access to a third-party LMS and there is no direct integration between the two. We have an old SAML integration using Identity Server 4 that is being replaced with Keycloak.

We need the user to log in to the portal. We need the user to click a link that automagically gets them the proper OAuth2 token for logging in to the LMS. We do not want the user to log in twice and don't want them to see any part of Keycloak in this flow. That's how it works in the ID4 SAML implementation.

I see I can connect to Keycloak via a client setup as a service account kind of login and get an access_token. I however cannot figure out how to use this to get information about a specific user nor generate a valid access_token for that user.  I don't see any advantage to providing the user with something that logs them in to Keycloak vs the server handing back a token directly.  We are using a Java portlet as the middle layer between the button and redirecting to the LMS.

**Comment by codespearhead** [comment]
The problem with what you have is that we can't easily reproduce it on own our ends, so we can't cut the corners and give you a concise step-by-step tutorial.

However, I pasted your question on Google's Gemini and it [responded with a decent roadmap](https://github.com/keycloak/keycloak/files/15156994/discussion-29097.md). Despite some outdated terms and the reference of deprecated Keycloak options, the essence of the steps are still valid in my opinion.

Try following it and if that doesn't work, let us know how you completed each step and what Keycloak version you're running.
